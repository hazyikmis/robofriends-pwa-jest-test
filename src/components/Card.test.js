//import { shallow, mount, render } from "enzyme";
import { shallow } from "enzyme";
//import toJson from 'enzyme-to-json';
import React from "react";
import Card from "./Card";

//console.log(toJson(shallow(<Card />))); //must logs out "ShallowWrapper {length: 1}"
//console.log(shallow(<Card />)); //must logs out "ShallowWrapper {length: 1}"

// it("expect to render Card component", () => {
//   expect(shallow(<Card />).length).toEqual(1);
// });

it("snapshot match testing: Card component", () => {
  //expect(shallow(<Card />).length).toEqual(1);
  //const wrapper = shallow(<Card />);
  
  const mockCard = {id: 1, name: "John Snow", email: "john@snow.com"}
  const wrapper = shallow(<Card id={mockCard.id} name={mockCard.name} email={mockCard.email} />);
  //expect(wrapper.exists()).toBe(true);
  expect(wrapper.debug()).toMatchSnapshot();
});
