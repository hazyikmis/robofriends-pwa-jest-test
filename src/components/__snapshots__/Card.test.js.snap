// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot match testing: Card component 1`] = `
"<div className=\\"tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5\\">
  <img alt=\\"robots\\" src=\\"https://robohash.org/1?size=200x200\\" />
  <div>
    <h2>
      John Snow
    </h2>
    <p>
      john@snow.com
    </p>
  </div>
</div>"
`;
